covenant-signer/itest/e2e_test.go

```diff
@@ -0,0 +1,352 @@
+		[]*btcec.PublicKey{covnenantPublicKey},
+		1,
+		stakingTime,
+		stakingAmount,
+		netParams,
+	)
+	require.NoError(t, err)
+
+	stakingTx := wire.NewMsgTx(2)
+	stakingTx.AddTxIn(fakeInput)
+	stakingTx.AddTxOut(stakingInfo.StakingOutput)
+
+	stakingSlashingSpendInfo, err := stakingInfo.SlashingPathSpendInfo()
+	require.NoError(t, err)
+	stakingUnbondingSpendInfo, err := stakingInfo.UnbondingPathSpendInfo()
+	require.NoError(t, err)
+
+	stakingSlashingScript := stakingSlashingSpendInfo.RevealedLeaf.Script
+	stakingUnbondingScript := stakingUnbondingSpendInfo.RevealedLeaf.Script
+
+	unbondingInfo, err := staking.BuildUnbondingInfo(
+		stakerPrivKey.PubKey(),
+		[]*btcec.PublicKey{finalityProviderKey.PubKey()},
+		[]*btcec.PublicKey{covnenantPublicKey},
+		1,
+		unbondingTime,
+		stakingAmount-unbondingFee,
+		netParams,
+	)
+	require.NoError(t, err)
+
+	unbondingSlashingSpendInfo, err := unbondingInfo.SlashingPathSpendInfo()
+	require.NoError(t, err)
+	unbondingSlashingScript := unbondingSlashingSpendInfo.RevealedLeaf.Script
+
+	stakingTxHash := stakingTx.TxHash()
+	stakingOutputIndex := uint32(0)
+
+	unbondingTx := wire.NewMsgTx(2)
+	unbondingTx.AddTxIn(wire.NewTxIn(wire.NewOutPoint(&stakingTxHash, stakingOutputIndex), nil, nil))
+	unbondingTx.AddTxOut(unbondingInfo.UnbondingOutput)
+
+	stakingSlashingTx, err := btcstaking.BuildSlashingTxFromStakingTxStrict(
+		stakingTx,
+		stakingOutputIndex,
+		stakingSlashingScript,
+		stakerPrivKey.PubKey(),
+		unbondingTime,
+		int64(slashingFee),
+		slashingRate,
+		netParams,
+	)
+	require.NoError(t, err)
+
+	unbondingSlashingTx, err := btcstaking.BuildSlashingTxFromStakingTxStrict(
+		unbondingTx,
+		0,
+		unbondingSlashingScript,
+		stakerPrivKey.PubKey(),
+		unbondingTime,
+		int64(slashingFee),
+		slashingRate,
+		netParams,
+	)
+	require.NoError(t, err)
+
+	fpEncKey, err := asig.NewEncryptionKeyFromBTCPK(finalityProviderKey.PubKey())
+	require.NoError(t, err)
+
+	return signerapp.ParsedSigningRequest{
+		StakingTx:               stakingTx,
+		SlashingTx:              stakingSlashingTx,
+		UnbondingTx:             unbondingTx,
+		SlashUnbondingTx:        unbondingSlashingTx,
+		StakingOutputIdx:        stakingOutputIndex,
+		SlashingScript:          stakingSlashingScript,
+		UnbondingScript:         stakingUnbondingScript,
+		UnbondingSlashingScript: unbondingSlashingScript,
+		FpEncKeys:               []*asig.EncryptionKey{fpEncKey},
+	}
+}
+
+func TestSigningTransactions(t *testing.T) {
+	tm := StartManager(t, 100)
+
+	dataToSign := buildDataToSign(t, tm.covenantPrivKey.PubKey())
+
+	sigs, err := signerservice.RequestCovenantSignaure(
+		context.Background(),
+		tm.SigningServerUrl(),
+		10*time.Second,
+		&dataToSign,
+	)
+
+	require.NoError(t, err)
+	require.NotNil(t, sigs)
+
+	err = tm.verifyResponse(sigs, &dataToSign)
+	require.NoError(t, err)
+}
+
+func (tm *TestManager) verifyResponse(resp *signerapp.ParsedSigningResponse, req *signerapp.ParsedSigningRequest) error {
```

Comment:
can we separate `TestManager` functionallity and test cases, so having `testmanager.go` and `e2etest.go`